worldedit-bukkit/src/main/java/com/boydti/fawe/bukkit/regions/plotsquared/FaweSchematicHandler.java

```diff
@@ -45,64 +51,120 @@
+                return;
+            }
+            BlockVector3 dimension = schematic.getClipboard().getDimensions();
+            final int WIDTH = dimension.getX();
+            final int LENGTH = dimension.getZ();
+            final int HEIGHT = dimension.getY();
+            // Validate dimensions
+            CuboidRegion region = plot.getLargestRegion();
+            if (((region.getMaximumPoint().getX() - region.getMinimumPoint().getX() + xOffset + 1) < WIDTH) || (
+                (region.getMaximumPoint().getZ() - region.getMinimumPoint().getZ() + zOffset + 1) < LENGTH) || (HEIGHT
+                > 256)) {
+                TaskManager.runTask(whenDone);
+                return;
+            }
+            // Calculate the optimal height to paste the schematic at
+            final int y_offset_actual;
+            if (autoHeight) {
+                if (HEIGHT >= 256) {
+                    y_offset_actual = yOffset;
+                } else {
+                    PlotArea pw = plot.getArea();
+                    if (pw instanceof ClassicPlotWorld) {
+                        y_offset_actual = yOffset + ((ClassicPlotWorld) pw).PLOT_HEIGHT;
+                    } else {
+                        y_offset_actual = yOffset + 1 + PlotSquared.platform().worldUtil()
+                            .getHighestBlockSynchronous(plot.getWorldName(), region.getMinimumPoint().getX() + 1,
+                                region.getMinimumPoint().getZ() + 1);
+                    }
+                }
+            } else {
+                y_offset_actual = yOffset;
+            }
 
-            ReadOnlyClipboard clipboard = ReadOnlyClipboard.of(editSession, region, false, true);
+            final BlockVector3 to = BlockVector3
+                .at(region.getMinimumPoint().getX() + xOffset, y_offset_actual, region.getMinimumPoint().getZ() + zOffset);
 
-            Clipboard holder = new BlockArrayClipboard(region, clipboard);
-            CompressedSchematicTag tag = new CompressedSchematicTag(holder);
-            whenDone.run(tag);
-        });
+            try (EditSession editSession = new EditSessionBuilder(FaweAPI.getWorld(plot.getWorldName())).checkMemory(false)
+                .fastmode(true).limitUnlimited().changeSetNull().autoQueue(false).build()) {
+                final Clipboard clipboard = schematic.getClipboard();
+                clipboard.paste(editSession, to, true, false, true);
+                if (whenDone != null) {
+                    whenDone.value = true;
+                    TaskManager.runTask(whenDone);
+                }
+            }
+        };
+        if (Fawe.isMainThread()) {
+            com.boydti.fawe.util.TaskManager.IMP.async(r);
+        } else {
+            r.run();
+        }
     }
 
     @Override
     public boolean save(CompoundTag tag, String path) {
         if (tag == null) {
-            PlotSquared.debug("&cCannot save empty tag");
+            logger.debug("&cCannot save empty tag");
```

Comment:
LOGGER.warn, debug is not used outside the debug option in the settings/config.
Also the Log4j knows nothing about ampersand color codes.